src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -67,9 +67,14 @@ public WorkItem(Test test, ITestFilter filter)
                 : ParallelScope.Default;
 
 #if !NETSTANDARD1_3 && !NETSTANDARD1_6
-            TargetApartment = Test.Properties.ContainsKey(PropertyNames.ApartmentState)
-                ? (ApartmentState)Test.Properties.Get(PropertyNames.ApartmentState)
-                : ApartmentState.Unknown;
+            if ((ApartmentState) (Test.Properties.Get(PropertyNames.ApartmentState) ?? ApartmentState.Unknown) == ApartmentState.Unknown &&
+                (ApartmentState) (Test.Parent?.Properties?.Get(PropertyNames.ApartmentState) ?? ApartmentState.Unknown) != ApartmentState.Unknown)
+            {
+                Test.Properties.Set(PropertyNames.ApartmentState, 
+                    Test.Parent?.Properties?.Get(PropertyNames.ApartmentState));
+            }
+
+            TargetApartment = (ApartmentState)(Test.Properties.Get(PropertyNames.ApartmentState) ?? ApartmentState.Unknown);
```

Comment:
@rinubaby The WorkItem is an artifact created each time the tests are run. It should not be changing any state of the test, only reading it.

Reply:
Sure, I will check how it broke,  based on what you are saying the fix should be at the time when tests are constructed using reflection, is that reasonable that the test "propertybag" should either have the apartment attribute set to a value other than "Unknown"  or get it from its parent if that has a value that is not "Unknown"?

Reply:
@rinubaby No, at least as far as I can tell, the Test is already correctly set up. You can check that in debug. The `ApartmentAttribute` seems to be always setting the property when the value is `ApartmentState.STA` or `ApartmentState.MTA` (it gives an error if you try to use `ApartmentState.Unknown` on the attribute).

When work items are constructed in `WorkItemBuilder` they are given a target apartment based on (1) the target apartment property of the test, if present or (2) the target apartment property (C# property, that is) of the parent work item, provided that the parent is other than `Unknown`.

What I would do to work this issue is first determine if the tests are correctly loaded. That is, the `TestAssembly` should be given `ApartmentState.STA`.

Then I would check to see if the work items are set up correctly. The one corresponding to the assembly should get `STA` and each child work item should pick up STA from its parent. One possibility is that the assembly work item is having it's property over-written between it's initial creation and the creation of it's child work items. This check should be pretty easy running Debug, since all work items are (supposedly) created by `WorkItemBuilder`.

What we want to avoid is masking the problem rather than figuring out and fixing its root cause.